### PR TITLE
use review period permissions in UI

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/components/menu/Menu.jsx [hunk 1]
```diff
@@ -9,9 +9,9 @@ import {
   selectHasAnniversaryReportPermission,
   selectHasBirthdayReportPermission,
   selectHasCheckinsReportPermission,
-  selectHasPulseReportPermission,
```
Comment (mvolkmann) [inline web-ui/src/components/menu/Menu.jsx:12]: This was renamed to `selectHasViewPulseReportPermission` for consistency with other permissions.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import PropTypes from 'prop-types';
 import queryString from 'query-string';
 import React, {
-  useEffect,
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:4]: Alphabetized imported items.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 3]
```diff
@@ -735,7 +742,7 @@ const TeamReviews = ({ onBack, periodId }) => {
             key={reviewer.id}
             label={reviewer.name}
             variant="outlined"
-            onDelete={() => deleteReviewer(member, reviewer)}
+            onDelete={canUpdate ? () => deleteReviewer(member, reviewer) : null}
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:745]: Lots of UI elements are not either hidden or disabled based on the values of `canUpdate` and `canLaunch`.

File: web-ui/src/context/selectors.js [hunk 4]
```diff
@@ -16,9 +16,20 @@ export const selectLoading = state => state.loading;
 export const selectReviewPeriods = state => state.reviewPeriods;
 export const selectPermissions = state => state.permissions;
 
+const hasPermission = permissionName => createSelector(
```
Comment (mvolkmann) [inline web-ui/src/context/selectors.js:19]: I removed a lot of code duplication by defining this function.

File: web-ui/src/context/selectors.js [hunk 5]
```diff
@@ -36,103 +47,58 @@ export const selectIsAdmin = createSelector(
...
-export const selectHasUpdateReviewAssignmentsPermission = createSelector(
-  selectUserProfile,
-  userProfile =>
-    userProfile &&
-    userProfile.role &&
-    userProfile.permissions.some(p =>
-      p?.permission?.includes('CAN_UPDATE_REVIEW_ASSIGNMENTS')
-    )
-);
+export const selectHasCreateReviewAssignmentsPermission =
```
Comment (mvolkmann) [inline web-ui/src/context/selectors.js:76]: I added selectors for all the new permissions related to review assignments and review periods.
Maybe Tim is removing some of these.

File: web-ui/src/pages/PermissionsPage.jsx [hunk 6]
```diff
@@ -69,6 +69,9 @@ const EditPermissionsPage = () => {
   const hasPermission = selectHasPermissionAssignmentPermission(state);
   const [selectedRole, setSelectedRole] = useState(roles[0]);
   const [categoriesList, setCategoriesList] = useState([]);
+  //TODO: This is not yet getting the "Pulse" category of permissions,
```
Comment (mvolkmann) [inline web-ui/src/pages/PermissionsPage.jsx:72]: Hopefully PR #2429 addresses this.

File: web-ui/src/pages/PulseReportPage.jsx [hunk 7]
```diff
@@ -38,7 +38,7 @@ import { AppContext } from '../context/AppContext.jsx';
 import {
   selectCsrfToken,
   selectCurrentUser,
-  selectHasPulseReportPermission,
+  selectHasViewPulseReportPermission,
```
Comment (mvolkmann) [inline web-ui/src/pages/PulseReportPage.jsx:41]: Just renamed this selector.

File: web-ui/src/components/reviews/TeamReviews.jsx (hunk 2 above)
Comment (jackkeller) [reply web-ui/src/components/reviews/TeamReviews.jsx:4]: My OCD thanks you for this!

File: web-ui/src/context/selectors.js (hunk 4 above)
Comment (jackkeller) [reply web-ui/src/context/selectors.js:19]: So much better to read!

**Review by jackkeller — APPROVED** [review]
🪄 of approval!

File: web-ui/src/context/selectors.js (hunk 5 above)
Comment (timyates) [reply web-ui/src/context/selectors.js:76]: I think all these are staying 👍

File: web-ui/src/pages/PulseReportPage.jsx [hunk 8]
```diff
@@ -244,10 +244,10 @@ const PulseReportPage = () => {
   };
 
   useEffect(() => {
-    //TODO: Skipping the permission check during testing
-    //      because this permission has not been implemented yet.
-    // if (selectHasPulseReportPermission(state)) loadPulses();
+    // TODO: Uncomment this check after PR #2429 is merged.
```
Comment (timyates) [inline web-ui/src/pages/PulseReportPage.jsx:247]: I just merged https://github.com/objectcomputing/check-ins/pull/2429

Comment (timyates) [reply web-ui/src/pages/PulseReportPage.jsx:247]: As far as I can tell, it's working 🎉 

Do we need a `PulseReportPage.test.jsx` though 🤔

Comment (mvolkmann) [reply web-ui/src/pages/PulseReportPage.jsx:253]: @timyates I added this test.

**Review by vhscom — APPROVED** [review]

File: web-ui/src/pages/PulseReportPage.jsx (hunk 8 above)
Comment (timyates) [reply web-ui/src/pages/PulseReportPage.jsx:253]: Pushed a fix for the test here https://github.com/objectcomputing/check-ins/pull/2479 the date was changing in the snapshot when under test

**Pull request merged**